lib/commons/dom/create-grid.js

```diff
@@ -324,11 +324,13 @@ function findScrollRegionParent(vNode, parentVNode) {
  */
 function addNodeToGrid(grid, vNode) {
   const gridSize = constants.gridSize;
-  // save a reference to where this element is in the grid so we
-  // can find it even if it's in a subgrid
-  vNode._grid = grid;
-
   vNode.clientRects.forEach(rect => {
+    if (rect.right <= 0 || rect.bottom <= 0) {
+      return;
+    }
+    // save a reference to where this element is in the grid so we
+    // can find it even if it's in a subgrid
+    vNode._grid ??= grid;
```

Comment:
IMPORTANT: These lines have been modified to fix a bug where elements outside the viewport could still be added to the grid.